src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormatInfoScanner.cs

```diff
@@ -109,37 +109,31 @@ internal class DateTimeFormatInfoScanner
         internal const string CJKMinuteSuff = "\u5206";
         internal const string CJKSecondSuff = "\u79d2";
 
-        // The collection fo date words & postfix.
+        // The collection for date words & postfix.
         internal List<string> m_dateWords = new List<string>();
         // Hashtable for the known words.
-        private static volatile Dictionary<string, string>? s_knownWords;
-
-        private static Dictionary<string, string> KnownWords =>
-            s_knownWords ??=
-            new Dictionary<string, string>(16)
-            {
-                // Add known words into the hash table.
-
-                // Skip these special symbols.
-                { "/", string.Empty },
-                { "-", string.Empty },
-                { ".", string.Empty },
-
-                // Skip known CJK suffixes.
-                { CJKYearSuff, string.Empty },
-                { CJKMonthSuff, string.Empty },
-                { CJKDaySuff, string.Empty },
-                { KoreanYearSuff, string.Empty },
-                { KoreanMonthSuff, string.Empty },
-                { KoreanDaySuff, string.Empty },
-                { KoreanHourSuff, string.Empty },
-                { KoreanMinuteSuff, string.Empty },
-                { KoreanSecondSuff, string.Empty },
-                { CJKHourSuff, string.Empty },
-                { ChineseHourSuff, string.Empty },
-                { CJKMinuteSuff, string.Empty },
-                { CJKSecondSuff, string.Empty }
-            };
+        private static readonly HashSet<string> s_knownWords = new HashSet<string>(16)
```

Comment:
I am wondering why we are using `HashSet<string>` here. All strings in it are single-character constants, so `HashSet<char>` or even a simple `switch (str[0])` would do the job.

Reply:
I didn't look at any of the strings, just that it was a dictionary that only cared about its key and not its value. If you're correct that they're all one character, we could delete this and change the single call site accordingly.